pilot/pkg/security/authz/matcher/header.go

```diff
@@ -36,22 +36,34 @@ func HeaderMatcher(k, v string) *routepb.HeaderMatcher {
 	} else if strings.HasPrefix(v, "*") {
 		return &routepb.HeaderMatcher{
 			Name: k,
-			HeaderMatchSpecifier: &routepb.HeaderMatcher_SuffixMatch{
-				SuffixMatch: v[1:],
+			HeaderMatchSpecifier: &routepb.HeaderMatcher_StringMatch{
+				StringMatch: &matcherpb.StringMatcher{
```

Comment:
The `StringMatcher` is added in Envoy recently (https://github.com/envoyproxy/envoy/pull/17119, the upcoming 1.20.0) and not available in older versions.

I think we should wait a little longer before using it (or guard it with version check) otherwise it will cause problems for older Envoy for in-place upgrade.

Reply:
you are right, it may be too early to update the `HeaderMatchSpecifier` to use `StringMatch` and that may cause compatibility issues with old version envoy